gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/reactor/handler/AccessPointHttpAcceptor.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.reactor.handler;
+
+import static java.util.stream.Collectors.toCollection;
+
+import io.gravitee.common.event.Event;
+import io.gravitee.common.event.EventListener;
+import io.gravitee.common.event.EventManager;
+import io.gravitee.gateway.api.Request;
+import io.gravitee.gateway.reactor.accesspoint.AccessPointEvent;
+import io.gravitee.gateway.reactor.accesspoint.ReactableAccessPoint;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * @author David BRASSELY (david.brassely at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class AccessPointHttpAcceptor implements HttpAcceptor, EventListener<AccessPointEvent, ReactableAccessPoint> {
+
+    private final String environmentId;
+    private final String path;
+    private final ReactorHandler reactor;
+    private final Collection<String> serverIds;
+    private List<HttpAcceptor> httpAcceptors;
+    private boolean isSingleHttpAcceptor;
+    private HttpAcceptor singleHttpAcceptor;
```

Comment:
Looks redundant. All the stuff you do with 'single...' can be done in a collection. Lots of conditions around that make the code harder to read.